metis-indexing/src/main/java/eu/europeana/indexing/TombstoneUtil.java

```diff
@@ -0,0 +1,214 @@
+package eu.europeana.indexing;
+
+import static java.util.function.Predicate.not;
+
+import eu.europeana.corelib.definitions.edm.entity.ChangeLog;
+import eu.europeana.corelib.definitions.edm.entity.EuropeanaAggregation;
+import eu.europeana.corelib.solr.bean.impl.FullBeanImpl;
+import eu.europeana.corelib.solr.entity.AgentImpl;
+import eu.europeana.corelib.solr.entity.AggregationImpl;
+import eu.europeana.corelib.solr.entity.ChangeLogImpl;
+import eu.europeana.corelib.solr.entity.ConceptImpl;
+import eu.europeana.corelib.solr.entity.EuropeanaAggregationImpl;
+import eu.europeana.corelib.solr.entity.LicenseImpl;
+import eu.europeana.corelib.solr.entity.OrganizationImpl;
+import eu.europeana.corelib.solr.entity.PlaceImpl;
+import eu.europeana.corelib.solr.entity.ProvidedCHOImpl;
+import eu.europeana.corelib.solr.entity.ProxyImpl;
+import eu.europeana.corelib.solr.entity.TimespanImpl;
+import eu.europeana.metis.utils.DepublicationReason;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.apache.commons.lang3.ArrayUtils;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Contains functionality for updating fields that are required for tombstone records.
+ */
+public class TombstoneUtil {
+
+  /**
+   * Prepare the tombstone fullbean.
+   *
+   * @param publishedFullbean the published fullbean
+   * @param depublicationReason the depublication reason
+   * @return the newly created tombstone
+   */
+  public FullBeanImpl prepareTombstoneFullbean(FullBeanImpl publishedFullbean, DepublicationReason depublicationReason) {
```

Comment:
my sugestion would be to use a `static` method and update the rest of private also to `static`